jraph/_src/utils.py

```diff
@@ -1122,3 +1123,33 @@ def wrapper(graph: gn_graph.GraphsTuple) -> gn_graph.GraphsTuple:
     return zero_out_padding(graph_net(graph))
 
   return wrapper
+
+
+def from_scipy_sparse_matrix(A: scipy.sparse.spmatrix):
```

Comment:
Yeah fair point, let me see what I can do

Reply:
Thanks for your patience while we think through this.

We're not sure this is general enough to go in the jraph utilities, but we can see the value of having this available. One option may to add it as an example, or add it to documentation, so that we can remove dependency requirements from the main library.

What do you think?

Reply:
I don't know why I did not think about this before. I think we can get away without importing `scipy` at all. We just have to be more explicit on which sparse matrix format we use. Fixing the COO sparse format should be enough where we expect users to pass in `senders`, `receivers`, `value` (for the number of edges between the sender and receiver) and `n_node`.

What do you think about this? Otherwise, I would be happy to add it as an example/documentation :)